domain/src/main/java/org/oppia/domain/question/QuestionDataController.kt

```diff
@@ -0,0 +1,83 @@
+package org.oppia.domain.question
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import org.oppia.app.model.Question
+import org.oppia.domain.topic.TEST_TOPIC_ID_0
+import org.oppia.domain.topic.TEST_TOPIC_ID_1
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProviders
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val QUESTION_DATA_PROVIDER_ID = "QuestionDataProvider"
+
+/** Controller for retrieving an exploration. */
+@Singleton
+class QuestionDataController @Inject constructor(
+  private val questionProgressController: QuestionProgressController,
+  private val dataProviders: DataProviders
+) {
+
+  /**
+   * Returns a list of [Question] objects given a topic ID.
```

Comment:
Should this indicate that it's returning all questions for a given topic ID?

Also, do we really need this method? If not, suggest removing it since we should rely on the progress controller to retrieve specific state during a practice session.

Reply:
Also, if we need this only for testing and not prod cases I suggest removing it & instead verify that calling begin/end below doesn't fail.

Reply:
Following up: I don't know what we would use this method for. The TopicController is used to retrieve the list of skills to display on the training page, so we should just be operating based on that list here.

Reply:
We still need to fetch a list of questions given topic id / list of skill ids right?

Reply:
I'm doing the fetch within startTrainingSession itself